src/lib/IP.coffee

```diff
@@ -0,0 +1,38 @@
+module.exports = class IP
+  @types: [
+    'data'
+    'openBracket'
+    'closeBracket'
+  ]
+
+  # Creates as new IP object
+  # Valid types: 'data', 'openBracket', 'closeBracket'
+  constructor: (@type = 'data', @data = null, options = {}) ->
+    @groups = [] # sync groups
+    @scope = null # sync scope id
+    @stream = null # substream id
+    @owner = null # packet owner process
+    for key, val of options
+      this[key] = val
+
+  # Creates a new IP copying its contents by value not reference
+  clone: ->
```

Comment:
We should add a `clonable` property that components could use to tell downstream that their packets are not cloning-safe